code/game/machinery/hologram.dm

```diff
@@ -41,313 +37,274 @@ var/const/HOLOPAD_MODE = RANGE_BASED
-			return
-		take_call(user)
+
+	if(incoming_connection)
+		audible_message("The pad hums quietly as it establishes a connection.")
+		take_call()
 		return
-	else if(caller_id && !incoming_connection && forced)
-		audible_message("Access denied. Terminating a command-level transmission locally is not permitted.")
+	else if(connected_pad)
+		if(forced)
+			audible_message("Access denied. Terminating a command-level transmission locally is not permitted.")
+			return
+		audible_message("Severing connection to distant holopad.")
+		end_call()
 		return
-	else if(caller_id && !incoming_connection)
-		visible_message("Severing connection to distant holopad.")
-		end_call(user)
+
+	if(last_request + 20 SECONDS > world.time)
+		to_chat(user, SPAN_WARNING("\The [src] is still cooling down since the last transmission."))
 		return
-	switch(alert(user,"Would you like to request an AI's presence or establish communications with another pad?", "Holopad","AI","Holocomms","Cancel"))
+
+	switch(alert(user, "Would you like to request an AI's presence or establish communications with another pad?", "Holopad", "AI", "Holocomms", "Cancel"))
 		if("AI")
-			if(last_request + 200 < world.time) //don't spam the AI with requests you jerk!
-				last_request = world.time
-				to_chat(user, "<span class='notice'>You request an AI's presence.</span>")
-				var/area/area = get_area(src)
-				for(var/mob/living/silicon/ai/AI in silicon_mob_list)
-					if(!AI.client)	continue
-					to_chat(AI, "<span class='info'>Your presence is requested at <a href='?src=\ref[AI];jumptoholopad=\ref[src]'>\the [area]</a>.</span>")
-			else
-				to_chat(user, "<span class='notice'>A request for AI presence was already sent recently.</span>")
+			last_request = world.time
+			to_chat(user, SPAN_NOTICE("You request an AI's presence."))
+			var/area/area = get_area(src)
+			for(var/mob/living/silicon/ai/AI in silicon_mob_list)
+				if(!AI.client)
+					continue
+				to_chat(AI, "<span class='info'>Your presence is requested at <a href='?src=\ref[AI];jumptoholopad=\ref[src]'>\the [area]</a>.</span>")
 		if("Holocomms")
-			if(user.loc != src.loc)
-				to_chat(user, "<span class='info'>Please step unto the holopad.</span>")
+			last_request = world.time
+			var/obj/item/card/id/I = user.GetIdCard()
+			if(!I)
+				to_chat(user, SPAN_NOTICE("You need authorization to use the holocall system. Please equip a valid ID card."))
+				return
+			var/forced_call = FALSE
+			if(access_heads in I.access) //Special functions for command level people
+				switch(alert(user,"Command level authorization detected. Additional functions available.", "Command Level Menu", "Forced Call", "Regular Call"))
+					if("Forced Call")
+						forced_call = TRUE
+			var/list/holopadlist = list()
+			for(var/obj/machinery/hologram/holopad/H in SSmachinery.processing_machines)
+				if(H != src && ARE_Z_CONNECTED(H.z, z) && H.operable())
```

Comment:
```suggestion
			for(var/obj/machinery/hologram/holopad/H in SSmachinery.processing_machines - src)
				if(ARE_Z_CONNECTED(H.z, z) && H.operable())
```

Could avoid doing the `H != src` check every time, but this isn't very impactful so it can go either way.